api/src/main/java/jakarta/data/Order.java

```diff
@@ -60,16 +60,16 @@
  * alphabetically by last name. Employees with the same salary and last name
  * are then sorted alphabetically by first name.</p>
  *
- * <p>A repository method will fail with a
- * {@link jakarta.data.exceptions.DataException DataException}
- * or a more specific subclass if</p>
- * <ul>
- * <li>an {@code Order} parameter is
- *     specified in combination with a {@link PageRequest} parameter with
- *     {@link PageRequest#sorts()}.</li>
- * <li>the database is incapable of ordering with the requested
- *     sort criteria.</li>
- * </ul>
+ * <p>A repository method may not be declared with more than one parameter
+ * of type {@code Order}.</p>
+ *
+ * <p>A repository method throws {@link IllegalArgumentException} if it is
+ * called with an argument of type {@code PageRequest} with nonempty sort
+ * criteria, and a separate argument of type {@link Order}.</p>
+ *
+ * <p>A repository method throws {@link UnsupportedOperationException} if
+ * the database is incapable of ordering the query results using the given
+ * sort criteria.</p>
```

Comment:
I was trying to think of why this was originally written as DataException (which is the generic exception used by Jakarta Data) and I realized that there are cases where a Jakarta Data provider won't automatically know if some entity attribute types are valid for sort criteria until it is tried on the database, and then at that point, will only know that some error was sent back by the database/Jakarta Persistence provider/other, but not be certain of the reason for the error.  Maybe it was due to a lack of support for sorting on that column, or maybe it was due to another error condition that is completely unrelated.

We could possibly address this by writing something like this,
```
 * <p>A repository method throws {@link UnsupportedOperationException} if
 * the Jakarta Data provider is able to detect that the database is incapable of
 * ordering the query results using the given sort criteria. Otherwise, if the cause
 * of the error might be an inability to apply the sort criteria or might be something
 * else, then the repository method raises {@link DataException}.</p>
```

but I suspect that is too complicated and would be best just to stick with DataException,
```suggestion
 * <p>A repository method throws {@link DataException} if the database
 * is incapable of ordering the query results using the given sort criteria.</p>
```